File: rest/src/main/java/io/sparkled/viewmodel/ViewModelConverter.java
package io.sparkled.viewmodel;

/**
 * Handles view model -> model conversion.
 *
 * @param <VM> A ViewModel class
 * @param <M>  A Model class
 */
public interface ViewModelConverter<VM extends ViewModel, M> {

    /**
     * @param viewModel The view model class to be converted
     * @return The converted model.
     */
    M toModel(VM viewModel);
}

File: rest/src/main/kotlin/io/sparkled/viewmodel/ViewModelConverter.kt
package io.sparkled.viewmodel

/**
 * Handles view model -> model conversion.

 * @param <VM> A view model class
 * *
 * @param <M>  A model class
</M></VM> */
interface ViewModelConverter<VM : ViewModel, M> {

    /**
     * @param viewModel The view model to be converted
     * *
     * @return The converted model.
     */
    fun toModel(viewModel: VM): M
}
